### PR TITLE
lightning: fix tidb-backend estimate data source

### PR DESCRIPTION
**Pull request by Little-Wallace**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/28276
Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Unit test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/restore_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/pingcap/tidb/table/tables"
```
Comment (kennytm) [inline br/pkg/lightning/restore/restore_test.go:21]: move this to line 64

File: br/pkg/lightning/restore/check_info.go [hunk 2]
```diff
@@ -397,12 +397,12 @@ func (rc *Controller) estimateSourceData(ctx context.Context) (int64, error) {
...
 		}
 		for _, tbl := range db.Tables {
+			originSource += tbl.TotalSize
 			tableInfo, ok := info.Tables[tbl.Name]
 			if ok {
-				originSource += tbl.TotalSize
 				// Do not sample small table because there may a large number of small table and it will take a long
 				// time to sample data for all of them.
-				if tbl.TotalSize < int64(config.SplitRegionSize) {
+				if rc.cfg.TikvImporter.Backend != config.BackendLocal || tbl.TotalSize < int64(config.SplitRegionSize) {
```
Comment (kennytm) [inline br/pkg/lightning/restore/check_info.go:405]: i think the sampling is useful for importer backend  too

```suggestion
				if rc.cfg.TikvImporter.Backend == config.BackendTiDB || tbl.TotalSize < int64(config.SplitRegionSize) {
```

**Review by kennytm — APPROVED** [review]
rest LGTM

File: br/pkg/lightning/restore/restore_test.go (hunk 1 above)
Comment (Little-Wallace) [reply br/pkg/lightning/restore/restore_test.go:21]: Done

**Review by gozssky — APPROVED** [review]

**Pull request merged**